fastly/block_fastly_service_logging_newrelicotlp_test.go

```diff
@@ -0,0 +1,246 @@
+		FormatVersion:  2,
+		Format:         appendNewLine(newrelicotlpDefaultFormat),
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccCheckServiceVCLDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceVCLNewRelicOTLPConfig(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceExists("fastly_service_vcl.foo", &service),
+					testAccCheckFastlyServiceVCLNewRelicOTLPAttributes(&service, []*gofastly.NewRelicOTLP{&log1}, ServiceTypeVCL),
+					resource.TestCheckResourceAttr(
+						"fastly_service_vcl.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_vcl.foo", "logging_newrelicotlp.#", "1"),
+				),
+			},
+
+			{
+				Config: testAccServiceVCLNewRelicOTLPConfigUpdate(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceExists("fastly_service_vcl.foo", &service),
+					testAccCheckFastlyServiceVCLNewRelicOTLPAttributes(&service, []*gofastly.NewRelicOTLP{&log1AfterUpdate, &log2}, ServiceTypeVCL),
+					resource.TestCheckResourceAttr(
+						"fastly_service_vcl.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_vcl.foo", "logging_newrelicotlp.#", "2"),
+				),
+				PreventDiskCleanup: true,
+			},
+		},
+	})
+}
+
+func testAccCheckFastlyServiceVCLNewRelicOTLPAttributes(service *gofastly.ServiceDetail, newrelicotlp []*gofastly.NewRelicOTLP, serviceType string) resource.TestCheckFunc {
+	return func(_ *terraform.State) error {
+		conn := testAccProvider.Meta().(*APIClient).conn
+		newrelicotlpList, err := conn.ListNewRelicOTLP(&gofastly.ListNewRelicOTLPInput{
+			ServiceID:      service.ID,
+			ServiceVersion: service.ActiveVersion.Number,
+		})
+		if err != nil {
+			return fmt.Errorf("error looking up NewRelic OTLP Logging for (%s), version (%d): %s", service.Name, service.ActiveVersion.Number, err)
+		}
+
+		if len(newrelicotlpList) != len(newrelicotlp) {
+			return fmt.Errorf("newRelic List count mismatch, expected (%d), got (%d)", len(newrelicotlp), len(newrelicotlpList))
+		}
+
+		log.Printf("[DEBUG] newrelicotlpList = %#v\n", newrelicotlpList)
+
+		var found int
+		for _, d := range newrelicotlp {
+			for _, dl := range newrelicotlpList {
+				if d.Name == dl.Name {
+					// we don't know these things ahead of time, so populate them now
+					d.ServiceID = service.ID
+					d.ServiceVersion = service.ActiveVersion.Number
+					// We don't track these, so clear them out because we also won't know
+					// these ahead of time
+					dl.CreatedAt = nil
+					dl.UpdatedAt = nil
+
+					// Ignore VCL attributes for Compute and set to whatever is returned from the API.
+					if serviceType == ServiceTypeCompute {
+						dl.FormatVersion = d.FormatVersion
+						dl.Format = d.Format
+						dl.ResponseCondition = d.ResponseCondition
+						dl.Placement = d.Placement
+					}
+
```

Comment:
This looks like a copy/paste mistake as it's been mentioned that Compute isn't supported for this resource.

```suggestion
```